### PR TITLE
Remove src directory, shuffle layout, enable reporting.

### PR DESCRIPTION
**Pull request by devec0**
Having all of the charm "meat" in the `src` directory prevents charms developed from this template from being used as submodules directly, for instance, when using a layer based on this template in the `layers` directory of a templated charm.

This change moves all files to the top level, retains the layers and interfaces directories at the top level, and adjusts the `Makefile` accordingly.

It also turns on coverage reporting in HTML and annotated source code formats automatically.

### CODE REVIEW

**Comment by chris-sanders** [comment]
+1 lgtm

File: Makefile [hunk 1]
```diff
@@ -22,29 +17,34 @@ submodules:
...
+	@-git describe --tags > ./repo-info
+	@LAYER_PATH=./layers INTERFACE_PATH=./interfaces TERM=linux \
+		JUJU_REPOSITORY=$(JUJU_REPOSITORY) charm build . --force
 
 release: clean build
 	@echo "Charm is built at $(JUJU_REPOSITORY)/builds"
 
+push: release
+	@echo "Pushing $(JUJU_REPOSITORY)/builds/gitlab"
+	@charm push $(JUJU_REPOSITORY)/builds/gitlab cs:~pirate-charmers/gitlab
```
Comment (chris-sanders) [inline Makefile]: You've left 'gitlab' hard coded this section needs to use template variables or be left for the charm to implement.

Comment (devec0) [reply Makefile]: Fixed

**Comment by chris-sanders** [comment]
lgtm

**Pull request merged**